packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.scss

```diff
@@ -6,216 +6,250 @@
-.root {
-  box-sizing: border-box;
-  color: $ms-color-neutralPrimary;
-  font-size: $ms-font-size-m;
-  font-weight: $ms-font-weight-regular;
-  min-height: 36px;
-  position: relative;
+.input {
+  margin: 0px;
+  height: 0px;
+  width: 0px;
+  opacity: 0;
 }
 
-.textLabel {
-  font-size: $ms-font-size-m;
-  @include padding(0, 0, 0, 26px);
-  display: inline-block;
-  font-size: $ms-font-size-m;
+.label {
+  display: flex;
+  align-items: center;
+  margin: 0 -4px;
+  cursor: pointer;
+
+  > * {
+    margin: 0 4px;
+  }
 }
 
-// The hidden input
-.input {
-  position: absolute;
-  opacity: 0;
-  top: 8px
+.box {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex: 0 0 $ms-checkbox-label-size;
+  height: $ms-checkbox-label-size;
 }
 
-// The checkbox square container
-.label::before {
-  content: '';
-  display: inline-block;
-  border: 1px solid $ms-color-neutralTertiary;
-  width: $ms-checkbox-label-size;
+.checkbox {
+  display: flex;
+  align-items: center;
+  justify-content: center;
   height: $ms-checkbox-label-size;
-  font-weight: normal;
-  position: absolute;
+  width: $ms-checkbox-label-size;
+  border: 1px solid transparent;
   box-sizing: border-box;
   transition-property: background, border, border-color;
   transition-duration: $ms-checkbox-transition-duration;
   transition-timing-function: $ms-checkbox-transition-timing;
-}
+  /* incase the icon is bigger than the box */
+  overflow: hidden;
 
-// The check mark icon
-.label::after {
-  @include ms-Icon--CheckMark;
-  font-family: 'FabricMDL2Icons';
-  display: none;
-  position: absolute;
-  font-weight: 900;
-  background-color: transparent;
-  font-size: $ms-font-size-s-plus;
-  top: 0;
-  color: $ms-color-white;
-  line-height: $ms-checkbox-label-size;
-  width: $ms-checkbox-label-size;
-  text-align: center;
-}
-
-// The checkbox label
-.label {
-  display: inline-block;
-  cursor: pointer;
-  margin-top: 8px;
-  position: relative;
-  vertical-align: top;
-  user-select: none;
-  min-width: $ms-checkbox-label-size;
-  min-height: $ms-checkbox-label-size;
-  line-height: $ms-checkbox-label-size;
-
-  &:hover {
-    &::before {
-      border-color: $ms-color-neutralSecondaryAlt;
-    }
+  @include ms-borderColor-neutralTertiary;
+  @media screen and (-ms-high-contrast: active) {
+    border-color: $ms-color-contrastBlackSelected;
+  }
 
-    .textLabel {
-      color: $ms-color-black;
-    }
+  @media screen and (-ms-high-contrast: black-on-white) {
+    border-color: $ms-color-contrastWhiteSelected;
   }
+}
 
-  &:focus {
-    &::before {
-      border-color: $ms-color-neutralSecondaryAlt;
-    }
+.checkmark {
+  opacity: 0;
+  flex: 0 0 auto;
+  @include ms-fontColor-white;
+}
 
-    &.labelIsDisabled::before {
-      border-color: $ms-color-neutralTertiaryAlt;
-    }
+.text {
+  flex: 1 0 auto;
+}
 
-    &.labelIsChecked::before {
-      border-color: $ms-color-themeDarkAlt;
-    }
+.root.isInFocus {
+  border: 1px solid $focusedBorderColor;
```

Comment:
Something very annoying here is that you don't have access to the browser's focus styling... so everything else looks normal except for this which looks entirely different.

Further, for apps like ours, we turn off outline as our PM's don't want them on mouse actions, only if a user tabs. Not sure this is the best solution here... I'm all ears for better options though!